### PR TITLE
update eslint config

### PR DESCRIPTION
**Pull request by tinomen**
fix linting issues (mostly with node-fetch use)
update the linting config
add a lint ci action

### CODE REVIEW

File: src/daos/Canvas/client.ts [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
+    };
   }
 
-  public async request(path: string = "", options = {}) {
-    const url = this.options.base_url + path
+  public async request(path = "", options: unknown = {}): Promise<any> {
+    const url = `${this.options.base_url}${path}`;
 
     const config = {
       ...options,
```
Comment (frankfusco) [inline src/daos/Canvas/client.ts:33]:
```suggestion
  public async request(path = "", options: Record<string, unknown> = {}): Promise<any> {
    const url = `${this.options.base_url}${path}`;

    const config = {
      ...options as {},
```

File: src/daos/Canvas/client.ts [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
-      headers: this.headers
-    }
+      headers: this.headers,
```
Comment (frankfusco) [inline src/daos/Canvas/client.ts:34]:
```suggestion
      headers: this.headers as {},
```

**Review by frankfusco — CHANGES_REQUESTED** [review]
TS is unhappy about `unknown`s with `fetch`'s `Request` type.

File: src/daos/Canvas/client.ts (hunk 1 above)
Comment (tinomen) [reply src/daos/Canvas/client.ts]: ok, Record has other issues as well. I'm redoing this to minimize type issues and the number of "any"s

**Review by frankfusco — APPROVED** [review]

**Review by frankfusco — APPROVED** [review]
Should we move the Canvas-specific stuff out of BaseClient after this?

**Pull request merged**